src/syntax/read.rkt

```diff
@@ -52,6 +52,7 @@
     [#`(/ #,a) (datum->syntax #f (list '/ 1 (expand a)) stx)]
     ; variary operators
     [#`(,(and (or '+ '- '* '/ 'or) op) #,arg1 #,arg2 #,rest ...)
+     (if (not (null? rest)) (warn 'variary-operator "~a is not supported as a variary operator" op) #f)
```

Comment:
Make the two argument case a separate match case. Please also rephrase a warming to say it is deprecated. It's still supported! Please use unless instead of if not. Also please warn for zero and one argument versions (except negation).